engine/src/main/java/org/camunda/bpm/engine/impl/dmn/entity/repository/DecisionRequirementsDefinitionQueryImpl.java

```diff
@@ -193,18 +194,24 @@ public DecisionRequirementsDefinitionQuery orderByTenantId() {
 
   @Override
   public long executeCount(CommandContext commandContext) {
-    checkQueryOk();
-    return commandContext
-      .getDecisionRequirementsDefinitionManager()
-      .findDecisionRequirementsDefinitionCountByQueryCriteria(this);
+    if (DecisionDefinitionQueryImpl.isDmnEnabled()) {
+      checkQueryOk();
+      return commandContext
+              .getDecisionRequirementsDefinitionManager()
+              .findDecisionRequirementsDefinitionCountByQueryCriteria(this);
+    }
+    return 0;
   }
 
   @Override
   public List<DecisionRequirementsDefinition> executeList(CommandContext commandContext, Page page) {
-    checkQueryOk();
-    return commandContext
-      .getDecisionRequirementsDefinitionManager()
-      .findDecisionRequirementsDefinitionsByQueryCriteria(this, page);
+    if (DecisionDefinitionQueryImpl.isDmnEnabled()) {
```

Comment:
❌ See above:
```suggestion
    if (commandContext.getProcessEngineConfiguration().isDmnEnabled()) {
```

Reply:
Thank you @yanavasileva for the suggestions. I've updated the code.